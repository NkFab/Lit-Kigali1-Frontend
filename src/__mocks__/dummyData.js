module.exports = {
  signupUser: {
    username: 'test',
    email: 'test@email.com',
    password: 'test@test',
    firstName: 'test',
    lastName: 'test',
  },
  articleData: {
    id: 'f9a4731d-7dea-4a9a-9b7d-8766d651c202',
    userId: 'dfef16f9-11a7-4eae-9ba0-7038c6ccaa73',
    slug: 'new-article-1',
    title: 'new article one',
    description: 'new article',
    body:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla faucibus ipsum non metus finibus ultricies. Donec ac auctor dui, sed fringilla est. Duis et pellentesque nisl, a gravida felis. Ut tempor felis id dignissim congue. Nunc blandit nunc sit amet dui pharetra, quis porttitor sem ullamcorper. Suspendisse faucibus imperdiet lacinia.',
    status: 'published',
    readingTime: '1 min',
    tagList: ['test', 'article'],
  },
  articleReducerState: {
    articles: [],
  },
  ratingData: {
    id: '1a322238-dadf-406c-8252-df926958622e',
    userId: 'd018c3b5-13c7-41c0-8b2c-4ec1cb6b21da',
    articleId: 'f9a4731d-7dea-4a9a-9b7d-8766d651c202',
    state: null,
    rating: 3,
    createdAt: '2019-04-19T14:14:09.096Z',
    updatedAt: '2019-04-19T17:50:48.826Z',
    author: {
      username: 'username',
      firstName: 'First Name',
      lastName: 'Last Name',
      image: null,
    },
    url: '/articles/article-slug/rating',
  },
  draftjsBody: [
    {
      key: 'dhjdh',
      text:
        'orem ipsum dolor sit amet, consectetur adipiscing elit. Nulla egestas justo arcu, sit amet eleifend velit venenatis et. Curabitur vel ipsum est. Phasellus iaculis arcu vel ligula faucibus laoree',
    },
    {
      key: 'loetvbd',
      text:
        'orem ipsum dolor sit amet, consectetur adipiscing elit. Nulla egestas justo arcu, sit amet eleifend velit venenatis et. Curabitur vel ipsum est. Phasellus iaculis arcu vel ligula faucibus laoree',
    },
    {
      key: 'ottsb',
      text:
        'orem ipsum dolor sit amet, consectetur adipiscing elit. Nulla egestas justo arcu, sit amet eleifend velit venenatis et. Curabitur vel ipsum est. Phasellus iaculis arcu vel ligula faucibus laoree',
    },
    {
      key: 'hsdghsd',
      text:
        'Maecenas a tellus sit amet metus congue dignissim at et ante. Etiam mollis sagittis elit sit amet vestibulum. In hac habitasse platea dictumst. Aliquam ut varius ante',
    },
  ],
  urlValue:
    'https://cdn.shopify.com/s/files/1/0550/6737/products/ORANGEWOOD_MANHATTAN_SPRUCE_BEGINNER_DREADNAUGHT_ACOUSTIC_GUITAR-1_2000x.png?v=1524595098',
  jsonFormat:
    '{"key":"dhjdh","text":"orem ipsum dolor sit amet, consectetur adipiscing elit. Nulla egestas justo arcu, sit amet eleifend velit venenatis et. Curabitur vel ipsum est. Phasellus iaculis arcu vel ligula faucibus laoree"}',
  entityMap1: {
    0: {
      type: 'LINK',
      mutability: 'MUTABLE',
      data: {
        url: 'https://www.lipsum.com/feed/html',
      },
    },
    1: {
      type: 'image',
      mutability: 'IMMUTABLE',
      data: {
        src:
          'https://res.cloudinary.com/litkigaliauthorshaven/image/upload/v1556089726/beach-exotic-holiday-248797_fegfqb.jpg',
      },
    },
    2: {
      type: 'LINK',
      mutability: 'MUTABLE',
      data: {
        url: 'https://www.lipsum.com/feed/html',
      },
    },
    3: {
      type: 'LINK',
      mutability: 'MUTABLE',
      data: {
        url: 'https://www.lipsum.com/feed/html',
      },
    },
    4: {
      type: 'video',
      mutability: 'IMMUTABLE',
      data: {
        src:
          'https://res.cloudinary.com/litkigaliauthorshaven/video/upload/v1556089741/BirdNoSound_1_klovmd.mp4',
      },
    },
  },
  entityMap2: {
    0: {
      type: 'LINK',
      mutability: 'MUTABLE',
      data: {
        url: 'https://www.lipsum.com/feed/html',
      },
    },
    2: {
      type: 'LINK',
      mutability: 'MUTABLE',
      data: {
        url: 'https://www.lipsum.com/feed/html',
      },
    },
    3: {
      type: 'LINK',
      mutability: 'MUTABLE',
      data: {
        url: 'https://www.lipsum.com/feed/html',
      },
    },
    4: {
      type: 'video',
      mutability: 'IMMUTABLE',
      data: {
        src:
          'https://res.cloudinary.com/litkigaliauthorshaven/video/upload/v1556089741/BirdNoSound_1_klovmd.mp4',
      },
    },
  },
  file: {
    lastModified: 1553947426997,
    lastModifiedDate: new Date(),
    name: 'beach-exotic-holiday-248797.jpg',
    size: 522195,
    type: 'image/jpeg',
    webkitRelativePath: '',
  },
  articleDataDraft: {
    id: 'f9a4731d-7dea-4a9a-9b7d-8766d651c202',
    userId: 'dfef16f9-11a7-4eae-9ba0-7038c6ccaa73',
    slug: 'new-article-1',
    title: 'new article one',
    description: 'The CEOs of YouTube and Twitter are being',
    body:
      '{"blocks":[{"key":"crkve","text":"The CEOs of YouTube and Twitter are being far more cautious in addressing the problems of their platforms than they were in building them","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
    status: 'published',
    readingTime: '1 min',
    tagList: ['test', 'article'],
  },
  article: {
    title: 'hello world',
    body: 'ejej lkejke lkejde eljfe kefnef efnefve lkenfefenf kefnerjfnern elkfnrfernflef elkfnejfernf lefnerfe kefnrefne klekfnerfefnlef ekfnefnerel nefefnef efnefklnef kenfefnef kenferfn',
    tagList: [],
  },
};
