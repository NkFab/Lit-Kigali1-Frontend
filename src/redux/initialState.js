module.exports = {
  user: {
    loading: true,
    profile: {
      articles: [],
    },
    articles: [],
    favorites: [],
  },
  forgotPassword: {
    submitting: false,
    success: false,
    successMessage: '',
    email: '',
    newPassword: '',
    confirmNewpassword: '',
    message: '',
    errors: [],
  },
  currentUser: {
    message: '',
    following: false,
    deletingArticle: false,
    isLoggedIn: localStorage.getItem('token') !== null,
    profile: {
      articles: [],
    },
    articles: [],
    favorites: [],
  },
  signUp: {
    submitting: false,
    success: false,
    successMessage: '',
    username: '',
    email: '',
    password: '',
    message: '',
    errors: [],
  },
  login: {
    submitting: false,
    credentials: {
      username: '',
      password: '',
    },
  },
  article: {
    loading: true,
    submitting: false,
    searching: false,
    success: true,
    singleArticle: {
      title: '',
      body: '',
      tagList: [],
    },
    createArticle: {
      title: '',
      body: '',
      tagList: [],
    },
    articlesList: [],
  },
};
