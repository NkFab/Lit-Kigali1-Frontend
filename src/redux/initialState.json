{
  "user": {
    "profile": {},
    "articles": [],
    "favorites": []
  },
  "forgotPassword": {
    "submitting": false,
    "success": false,
    "successMessage": "",
    "email": "",
    "newPassword": "",
    "confirmNewpassword": "",
    "message": "",
    "errors": []
  },
  "currentUser": {
    "isLoggedIn": false,
    "profile": {},
    "articles": [],
    "favorites": []
  },
  "login": {
    "submitting": false,
    "credentials": {
      "username": "",
      "password": ""
    }
  }
}
