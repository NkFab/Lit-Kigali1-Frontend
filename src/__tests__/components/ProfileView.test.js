import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import {
  ProfileView,
  mapStateToProps,
  mapDispatchToProps,
} from '../../components/Profile/ProfileView';
import initialState from '../../redux/initialState';
import { articleData } from '../../__mocks__/dummyData';

let wrapper;
let store;
const mockStore = configureMockStore();
const props = {
  loading: true,
  profile: {
    articles: [],
  },
  currentUser: {
    username: 'username',
  },
  getUserProfile: jest.fn().mockImplementation(() => Promise.resolve({ status: 200 })),
  onFollowUser: jest.fn().mockImplementation(() => Promise.resolve({ status: 200 })),
  history: { push: jest.fn() },
};
describe('<ProfileView />', () => {
  beforeEach(() => {
    wrapper = mount(<ProfileView {...props} />);
  });
  test('should render the <ResetPassword />', () => {
    const renderedValue = renderer.create(<ProfileView {...props} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });

  test('should return empty div on loading', () => {
    wrapper.setProps({ laoding: true });
    expect(wrapper.props().loading).toBeTruthy();
  });

  test('should render edit button', () => {
    const user = { profile: { articles: [], username: 'username' } };
    wrapper.setProps({ profile: user, currentUser: user });
    expect(wrapper.props().currentUser.username).toEqual(wrapper.props().profile.username);
    expect(wrapper.instance().renderUserAction()).toBeDefined();
  });

  test('should render articles', () => {
    const newProps = {
      ...props,
      loading: false,
      profile: {
        articles: [articleData],
      },
    };
    wrapper = mount(<ProfileView {...newProps} />);
    expect(wrapper.instance().renderArticles()).toBeDefined();
  });

  test('should set following to true', () => {
    wrapper.setProps({ following: true, loading: false });
    expect(wrapper.props().following).toBeTruthy();
    expect(wrapper.find('.profile-meta__button').hasClass('loading'));
  });

  test('should render profile bio div', () => {
    wrapper.setProps({ profile: { articles: [], bio: 'My Bio' }, loading: false });
    expect(wrapper.props().profile.bio).toBe('My Bio');
  });

  describe('when clicking on submit button', () => {
    beforeEach(() => {
      store = mockStore({});
      wrapper = mount(<ProfileView store={store} {...props} />);
    });

    test('should call onFollowButton to `follow` a user', () => {
      wrapper.setProps({ loading: false });
      wrapper.find('.profile-meta__button').simulate('click');
      expect(props.onFollowUser).toHaveBeenCalled();
    });

    test('should call onFollowButton to `unfollow` a user', () => {
      wrapper.setProps({ loading: false, profile: { followed: true, articles: [] } });
      wrapper.find('.profile-meta__button').simulate('click');
      expect(props.onFollowUser).toHaveBeenCalled();
    });
  });

  describe('reducers', () => {
    test('should initialize the component state', () => {
      const state = mapStateToProps(initialState);
      expect(state).toHaveProperty('loading');
      expect(state).toHaveProperty('profile');
      expect(state).toHaveProperty('following');
      expect(state).toHaveProperty('currentUser');
    });
  });

  describe('actions creators', () => {
    test('should call getUserProfile action', () => {
      const payload = 'username';
      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).getUserProfile(payload);
      expect(dispatch).toHaveBeenCalled();
    });

    test('should call onFollowUser action', () => {
      const payload = { username: 'username', method: 'POST' };

      const dispatch = jest.fn();
      mapDispatchToProps(dispatch).onFollowUser(payload);
      expect(dispatch).toHaveBeenCalled();
    });
  });
});
