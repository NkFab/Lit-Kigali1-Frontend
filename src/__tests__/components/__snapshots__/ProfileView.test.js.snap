// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileView /> should render the <ResetPassword /> 1`] = `<div />`;
